maat-court-data-api/src/test/java/gov/uk/courtdata/integration/link/service/CreateLinkListenerIntegrationTest.java

```diff
@@ -85,7 +87,11 @@ public void givenSaveAndLinkModel_whenSaveAndImplIsInvoked_thenLinkEstablished()
         String saveAndLinkMessage = testModelDataBuilder.getSaveAndLinkString();
 
         //when
-        createLinkListener.receive(saveAndLinkMessage, "2");
+        Map<String, Object> header = new HashMap<>();
+        header.put("MessageId", "AIDAIU3GACVJITZULQ2RQ");
+        MessageHeaders headers = new MessageHeaders(header);
+        //when
+        createLinkListener.receive(saveAndLinkMessage, headers);
 
         //then
         CourtDataDTO courtDataDTO = testModelDataBuilder.getSaveAndLinkModelRaw();
```

Comment:
I think it is worth expanding the link integration tests to explicitly cover a scenario with a validation error. At the moment we only cover the happy path.